opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -257,12 +259,20 @@ public SpanBuilder withTag(String key, Number value) {
         }
 
         @Override
-        public SpanBuilder withStartTimestamp(long microseconds) {
-            this.startMicros = microseconds;
+        @Deprecated
+        public Tracer.SpanBuilder withStartTimestamp(long microseconds) {
+            return withStartTimestamp(microseconds, TimeUnit.MICROSECONDS);
+        }
+
+        @Override
+        public SpanBuilder withStartTimestamp(long startTimestamp, TimeUnit startUnit) {
+            this.startTimestamp = startTimestamp;
+            this.startUnit = startUnit;
             return this;
         }
 
         @Override
+        @Deprecated
```

Comment:
Why was `@Deprecated` added here?

Reply:
The interface which defines the method that is being overridden declares it as @Deprecated.  Not annotating this class still results in a compile-time warning and IDE static inspection flagging.